src/Message/Translator/RecalculatePlayheadTranslator.php

```diff
@@ -5,26 +5,52 @@
 namespace Patchlevel\EventSourcing\Message\Translator;
 
 use Patchlevel\EventSourcing\Aggregate\AggregateHeader;
+use Patchlevel\EventSourcing\Aggregate\StreamNameTranslator;
+use Patchlevel\EventSourcing\Message\HeaderNotFound;
 use Patchlevel\EventSourcing\Message\Message;
+use Patchlevel\EventSourcing\Store\StreamHeader;
 
 use function array_key_exists;
 
 final class RecalculatePlayheadTranslator implements Translator
 {
-    /** @var array<string, array<string, positive-int>> */
+    /** @var array<string, positive-int> */
     private array $index = [];
 
     /** @return list<Message> */
     public function __invoke(Message $message): array
     {
-        $header = $message->header(AggregateHeader::class);
-        $playhead = $this->nextPlayhead($header->aggregateName, $header->aggregateId);
+        try {
+            $header = $message->header(AggregateHeader::class);
+        } catch (HeaderNotFound) {
+            try {
+                $header = $message->header(StreamHeader::class);
+            } catch (HeaderNotFound) {
+                return [$message];
+            }
+        }
```

Comment:
This is a change in the behaviour of this translator. Before, we crashed here if the header was not present and now we are returning a valid message. Is this fine?

Reply:
I think it's OK. I find it strange when it crashes when no suitable header is found.

In this case, it's supposed to "fix" the playhead. But if there's no playhead, or no header with a playhead, then it has nothing to fix. Why should it stop here?

Reply:
I'm not sure anymore why we decided that tbh. But i think we had a reason. I agree with you, if nothing is there, then nothing to fix. I guess we implemented it that way, because a playhead was necessary. So, i think this is fine now. 👍